app/views/posts/index.html.erb

```diff
@@ -0,0 +1,25 @@
+<div class ="container">
+
+<% @posts.each do |post| %>
+
+ <div class="card my-2">
+  <div class="card-header">
+     <% if user_signed_in? %>
+    
+  By :  <%= post.user.name %> <br>
+  
+  <% else %>
+
+  By : Secret <br>
+<% end %>
```

Comment:
That is what we called logic in views, if/else.
Please, use **views helpers** to handle logic in views